### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by EntenKoeniq**
A nicer README.md file with clearer examples.

1. List of tested os added.
1. Main functions added.
1. "http::Request request{"http://test.com/test"};" changed to "http::Request request("http://test.com/test");".
1. "\n" was changed to "std::endl".

You can take a look here:  https://github.com/EntenKoeniq/HTTPRequest

## Update 2
1. Revert from "std::endl" to "\n".
1. Revert to using curly brace initialization.
1. "Linux" was changed to "GNU/Linux".
1. Android, BSD, and Haiku added.

## Update 3
1. Added missing curly brackets.

### CODE REVIEW

**Comment by elnormous** [comment]
I agree about the first two.
3. Curly brace initialization is preferred to function initializatio (see The C++ Programming Language 4th Edition or this article by Adrien Hamelin: https://isocpp.org/blog/2016/05/quick-q-why-is-list-initialization-using-curly-braces-better-than-the-alter).
4. "\n" is preferred to std::endl (please see C++ core guidelines: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rio-endl).

File: README.md [hunk 1]
```diff
@@ -1,65 +1,109 @@
...
+
+  #include "HTTPRequest.hpp"
+  
+  int main(void)
+  {
+      try
+      {
+          // create a request
+          // you can pass http::InternetProtocol::V6 to Request to make an IPv6 request
+          http::Request request("http://test.com/test");
```
Comment (elnormous) [inline README.md:28]: Revert to using curly brace initialization

File: README.md [hunk 2]
```diff
@@ -1,65 +1,109 @@
...
 
-HTTPRequest is a single-header library for making HTTP requests. You can just include it in your project and use it. HTTPRequest was tested on macOS, Windows and Linux, but it should work on most of the Linux-based platforms. Supports IPv4 and IPv6.
+HTTPRequest is a single-header library for making HTTP requests. You can just include it in your project and use it.
+| :heavy_check_mark: Supports IPv4 and IPv6 |
+| --- |
+
+# HTTPRequest was tested on
+- MacOS
+- Windows
+- Linux
```
Comment (elnormous) [inline README.md:10]: Add Android, BSD, and Haiku

File: README.md [hunk 3]
```diff
@@ -1,65 +1,109 @@
...
+
+          // send a get request
+          const auto response = request.send("GET");
+
+          // print the result
+          std::cout << std::string{response.body.begin(), response.body.end()} << std::endl;
+      }
+      catch (const std::exception& e)
+      {
+          std::cerr << "Request failed, error: " << e.what() << std::endl;
```
Comment (elnormous) [inline README.md:38]: Revert to "\n"

File: README.md (hunk 2 above)
Comment (elnormous) [inline README.md:10]: If you are listing operating systems and not kernels, then use "GNU/Linux" instead of "Linux"

Comment (EntenKoeniq) [reply README.md:10]: This has been taken from the current README.md "HTTPRequest was tested on macOS, Windows and Linux, but it should work on most of the Linux-based platforms."

I will change it to "GNU / Linux".

**Comment by EntenKoeniq** [comment]
1. Revert from "std::endl" to "\n".
1. Revert to using curly brace initialization.
1. "Linux" was changed to "GNU/Linux".
1. Android, BSD, and Haiku added.

**Pull request closed without merging**